File: lovelia/Tx.pl
#!/usr/bin/perl

use strict;
use warnings;

use FindBin;
use lib "$FindBin::Bin/lib";
use Userapi;
use account;
use actor;

sub Create {
my ($id, $user, $data) = @_;

eval {
my $hashtx = Userapi->transact(
{
actions => [
{
account => account,
name => "create",
authorization => [
{
actor => actor,
permission => "active",
},
],
data => {
id => $id,
user => $user,
data => $data,
},
},
],
},
{ broadcast => 1, sign => 1 }
);

bash
Copy code
print $hashtx, "\n";
};
if ($@) {
print $@, "\n";
}
}

sub Read {
my ($id) = @_;

eval {
my $hashId = Userapi->transact(
{
actions => [
{
account => account,
name => "read",
authorization => [
{
actor => actor,
permission => "active",
},
],
data => {
id => $id,
},
},
],
},
{ broadcast => 1, sign => 1 }
);
print $hashId, "\n";
};
if ($@) {
print $@, "\n";
}
}

sub Push {
my ($HasId, $user, $data) = @_;

Create($HasId, $user, $data);
Read($HasId);
}

Push(10543230, account, "inery task 4");

File: demigohu/tx.py
import os
import json
import datetime as dt
import pytz
from dotenv import load_dotenv
from api.cline import Cline
from api import keys
from api.utils import *
from urllib3.exceptions import InsecureRequestWarning
import warnings

warnings.filterwarnings("ignore", category=InsecureRequestWarning)


def node(id):
    load_dotenv()

    transact = Cline(url='')
    inery_account = os.environ["INERY_ACCOUNT"]
    p_keys = os.environ["PRIVATE_KEY"]
    x = {"id": id, "user": inery_account, "data": f"data {id}"}
    y = {"id": id}
    z = keys.INRKey(p_keys)

    bC = {
        "account": inery_account,
        "name": "create",
        "authorization": [{"actor": inery_account, "permission": "active"}],
        "data": transact.abi_json_to_bin(inery_account, "create", x)["binargs"]
    }
    bR = {
        "account": inery_account,
        "name": "read",
        "authorization": [{"actor": inery_account, "permission": "active"}],
        "data": transact.abi_json_to_bin(inery_account, "read", y)["binargs"]
    }
    transactionC = {
        "actions": [bC],
        "expiration": str((dt.datetime.utcnow() + dt.timedelta(seconds=60)).replace(tzinfo=pytz.UTC))
    }
    transactionR = {
        "actions": [bR],
        "expiration": str((dt.datetime.utcnow() + dt.timedelta(seconds=60)).replace(tzinfo=pytz.UTC))
    }

    getxC = transact.push_transaction(transactionC, z, broadcast=True)
    print("IIIIIIIIIIIIIIIIIIIIIIIIIIII")
    print("IIII Create Transaction IIII")
    print("IIIIIIIIIIIIIIIIIIIIIIIIIIII")
    print(json.dumps(getxC, indent=1))

    getxR = transact.push_transaction(transactionR, z, broadcast=True)
    print("IIIIIIIIIIIIIIIIIIIIIIIIIIII")
    print("IIIII Read Transaction IIIII")
    print("IIIIIIIIIIIIIIIIIIIIIIIIIIII")
    print(json.dumps(getxR, indent=1))


if __name__ == "__main__":
    node(1)
